src/multivariate.jl

```diff
@@ -90,17 +90,22 @@ end
 
 A multivariate Normal distribution whose covariance is dense. Compatible with Tracker.
 """
-struct TuringDenseMvNormal{Tm<:AbstractVector, TC<:Cholesky} <: ContinuousMultivariateDistribution
+struct TuringDenseMvNormal{Tm<:AbstractVector, TC<:AbstractMatrix} <: ContinuousMultivariateDistribution
     m::Tm
     C::TC
 end
 function TuringDenseMvNormal(m::AbstractVector, A::AbstractMatrix)
-    return TuringDenseMvNormal(m, cholesky(A))
+    return TuringDenseMvNormal(m, cholesky(A).L)
 end
+
+function TuringSqrtDenseMvNormal(m::AbstractVector, C::AbstractMatrix)
+    TuringDenseMvNormal(m, C)
+end
+
 Base.length(d::TuringDenseMvNormal) = length(d.m)
 Distributions.rand(d::TuringDenseMvNormal, n::Int...) = rand(Random.GLOBAL_RNG, d, n...)
 function Distributions.rand(rng::Random.AbstractRNG, d::TuringDenseMvNormal, n::Int...)
-    return d.m .+ d.C.U' * adapt_randn(rng, d.m, length(d), n...)
+    return d.m .+ d.C * adapt_randn(rng, d.m, length(d), n...)
```

Comment:
Here `.U'` is used on purpose to avoid a copy. By default `cholesky` computes the upper triangular matrix which can be accessed without any additional allocations - however, `.L` (as in the modified) creates a new array:
https://github.com/JuliaLang/julia/blob/719c4d01b98c9c6a0d531047a7d530eae90e2af9/stdlib/LinearAlgebra/src/cholesky.jl#L445-L457

Reply:
Or one can just call `cholesky(A, :L)` in the first place https://github.com/JuliaLang/julia/blob/719c4d01b98c9c6a0d531047a7d530eae90e2af9/stdlib/LinearAlgebra/src/cholesky.jl#L406

Reply:
Yes, that could be changed. It might have to be changed in other places as well though and would require to always use `.L` then. I guess it will be surprising if the defaults are different from Julia base - suddenly users will see additional allocations when they use the default factorization (e.g., by computing it yourself with PositiveFactorizations).